library/src/main/java/com/novoda/downloadmanager/lib/DownloadService.java

```diff
@@ -154,6 +157,13 @@ public void onCreate() {
         mExecutor = factory.createExecutor();
     }
 
+    private DownloadClientReadyChecker getDownloadClientReadyChecker() {
+        if (!(getApplication() instanceof DownloadClientReadyChecker)) {
```

Comment:
Can't you pass this into the constructor of the DownloadManager? Having a hard dependency on the Application class isn't very extensible